unified-runtime/source/common/logger/ur_logger_details.hpp

```diff
@@ -19,111 +19,84 @@ struct LegacyMessage {
-  }
-
-  template <typename... Args> void warn(const char *format, Args &&...args) {
-    warning(format, std::forward<Args>(args)...);
-  }
-
-  template <typename... Args> void error(const char *format, Args &&...args) {
-    log(logger::Level::ERR, format, std::forward<Args>(args)...);
-  }
-
-  template <typename... Args> void always(const char *format, Args &&...args) {
-    if (sink) {
-      sink->log(logger::Level::QUIET, format, std::forward<Args>(args)...);
-    }
-  }
-
-  template <typename... Args>
-  void debug(const logger::LegacyMessage &p, const char *format,
-             Args &&...args) {
-    log(p, logger::Level::DEBUG, format, std::forward<Args>(args)...);
-  }
-
   template <typename... Args>
-  void info(const logger::LegacyMessage &p, const char *format,
-            Args &&...args) {
-    log(p, logger::Level::INFO, format, std::forward<Args>(args)...);
-  }
-
-  template <typename... Args>
-  void warning(const logger::LegacyMessage &p, const char *format,
-               Args &&...args) {
-    log(p, logger::Level::WARN, format, std::forward<Args>(args)...);
-  }
-
-  template <typename... Args>
-  void error(const logger::LegacyMessage &p, const char *format,
-             Args &&...args) {
-    log(p, logger::Level::ERR, format, std::forward<Args>(args)...);
-  }
-
-  template <typename... Args>
-  void log(logger::Level level, const char *format, Args &&...args) {
-    log(logger::LegacyMessage(format), level, format,
+  void log(Level level, const char *filename, const char *lineno,
+           const char *format, Args &&...args) {
+    log(LegacyMessage(format), level, filename, lineno, format,
         std::forward<Args>(args)...);
   }
 
   template <typename... Args>
-  void log(const logger::LegacyMessage &p, logger::Level level,
-           const char *format, Args &&...args) {
+  void log(const LegacyMessage &p, Level level, const char *filename,
+           const char *lineno, const char *format, Args &&...args) {
     if (!sink) {
       return;
     }
 
     if (isLegacySink) {
-      sink->log(level, p.message, std::forward<Args>(args)...);
+      sink->log(level, filename, lineno, p.message,
+                std::forward<Args>(args)...);
       return;
     }
     if (level < this->level) {
       return;
     }
 
-    sink->log(level, format, std::forward<Args>(args)...);
+    sink->log(level, filename, lineno, format, std::forward<Args>(args)...);
   }
 
-  void setLegacySink(std::unique_ptr<logger::Sink> legacySink) {
+  void setLegacySink(std::unique_ptr<Sink> legacySink) {
     this->isLegacySink = true;
     this->sink = std::move(legacySink);
   }
 
 private:
-  logger::Level level;
-  std::unique_ptr<logger::Sink> sink;
+  Level level;
+  std::unique_ptr<Sink> sink;
   bool isLegacySink = false;
 };
 
 } // namespace logger
 
+#ifdef SRC_PATH_SIZE
+#define SHORT_FILE ((__FILE__) + (SRC_PATH_SIZE))
+#else
+#define SHORT_FILE __FILE__
+#endif
+
+#define UR_STRIMPL(x) #x
```

Comment:
Since this is an implementation macro, should it be suffixed with an underscore as well? i.e. `UR_STRIMPL_` and `UR_STR_`